### PR TITLE
FT-4 sync with gitlab, pull from gitlab, upload tokens, used @gitbreak/bro…

### PR DESCRIPTION
**Pull request by swordEdge**
…wser as a gitlab client

### CODE REVIEW

**Review by six7 — COMMENTED** [review]
thought i'd give you some comments while you're still working on this. this looks great so far! 🎉

File: src/app/components/SyncSettings.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const SyncSettings = () => {
     setShowEditStorageModalVisible(true);
   };
 
-  const selectedRemoteProvider = () => [StorageProviderType.JSONBIN, StorageProviderType.GITHUB, StorageProviderType.URL].includes(
+  const selectedRemoteProvider = () => [StorageProviderType.JSONBIN, StorageProviderType.GITHUB, StorageProviderType.GITLAB, StorageProviderType.URL].includes(
```
Comment (LiamMartens) [inline src/app/components/SyncSettings.tsx:38]: Could we just update this with a `useMemo` hook whilst we're working on this.
[useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)

File: src/app/components/SyncSettings.tsx [hunk 2]
```diff
@@ -175,6 +191,21 @@ const SyncSettings = () => {
                 text="GitHub"
                 id={StorageProviderType.GITHUB}
               />
+              <ProviderSelector
+                isActive={localApiState?.provider === StorageProviderType.GITLAB}
+                isStored={storageType?.provider === StorageProviderType.GITLAB}
+                onClick={() => {
```
Comment (LiamMartens) [inline src/app/components/SyncSettings.tsx:197]: Let's make use of React's useCallback hook so we don't create a new function reference every cycle

**Review by LiamMartens — CHANGES_REQUESTED** [review]
Left 2 comments

**Comment by swordEdge** [comment]
@six7 @LiamMartens 

> Errors on Node.js CI / Create Package, Node.js CI / Tests

The dependency(gitbeaker/browser) is relying on Node.js 16+, but Vercel currently does not have support yet, as we are dependent on AWS.
https://github.com/vercel/community/discussions/37

**Comment by six7** [comment]
> 

I think it's fine. Vercel is only required for previews, we should not make the plugin dependable on that.

Any updates on the failings tests? Would love to get this merged.

**Review by LiamMartens — APPROVED** [review]

**Comment by LiamMartens** [comment]
@swordEdge just need to update the`node.js.yml` workflow to use Node 14 and we should be all good

**Pull request merged**